data-prepper-plugins/geoip-processor/src/integrationTest/java/org/opensearch/dataprepper/plugins/geoip/processor/GeoIPProcessorIT.java

```diff
@@ -0,0 +1,255 @@
+import static org.opensearch.dataprepper.plugins.geoip.processor.GeoIPProcessor.GEO_IP_EVENTS_FAILED;
+import static org.opensearch.dataprepper.plugins.geoip.processor.GeoIPProcessor.GEO_IP_EVENTS_FAILED_ENGINE_EXCEPTION;
+import static org.opensearch.dataprepper.plugins.geoip.processor.GeoIPProcessor.GEO_IP_EVENTS_FAILED_IP_NOT_FOUND;
+import static org.opensearch.dataprepper.plugins.geoip.processor.GeoIPProcessor.GEO_IP_EVENTS_PROCESSED;
+import static org.opensearch.dataprepper.plugins.geoip.processor.GeoIPProcessor.GEO_IP_EVENTS_SUCCEEDED;
+
+@ExtendWith(MockitoExtension.class)
+public class GeoIPProcessorIT {
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper(new YAMLFactory().enable(YAMLGenerator.Feature.USE_PLATFORM_LINE_BREAKS))
+            .registerModule(new JavaTimeModule());
+    private static final String COUNTRY_URL ="https://geoip.maps.opensearch.org/v1/mmdb/geolite2-country/manifest.json";
+    private static final String CITY_URL = "https://geoip.maps.opensearch.org/v1/mmdb/geolite2-city/manifest.json";
+    private static final String ASN_URL = "https://geoip.maps.opensearch.org/v1/mmdb/geolite2-asn/manifest.json";
+    private static GeoIPProcessorConfig geoipProcessorConfig;
+    private static GeoIpConfigSupplier defaultGeoIpConfigSupplier;
+    @Mock
+    private PluginMetrics pluginMetrics;
+    @Mock
+    private ExpressionEvaluator expressionEvaluator;
+    @Mock
+    private Counter geoIpEventsProcessed;
+    @Mock
+    private Counter geoIpEventsFailed;
+    @Mock
+    private Counter geoIpEventsSucceeded;
+    @Mock
+    private Counter geoIpEventsFailedEngineException;
+    @Mock
+    private Counter geoIpEventsFailedIPNotFound;
+
+    @BeforeAll
+    public static void downloadDatabases() throws JsonProcessingException {
+        String geoipProcessorConfigYaml = "          entries:\n" +
+                "            - source: \"peer/ip\"\n" +
+                "              target: \"geo\"\n" +
+                "              include_fields: [\"country_name\", \"continent_name\", \"location\", \"asn\"]\n" +
+                "          geoip_when: '/peer/status == \"success\"'\n" +
+                "          tags_on_no_valid_ip: [\"private_ip\", \"invalid_ip\"]";
+        String geoipServiceConfigYaml = "        maxmind:\n" +
+                "            databases:\n" +
+                "              country: " + COUNTRY_URL + "\n" +
+                "              city: " + CITY_URL + "\n" +
+                "              asn: " + ASN_URL + "\n" +
+                "            cache_count: 1024\n" +
+                "            database_refresh_interval: P3D";
+
+        geoipProcessorConfig = OBJECT_MAPPER.readValue(geoipProcessorConfigYaml, GeoIPProcessorConfig.class);
+        GeoIpServiceConfig geoipServiceConfig = OBJECT_MAPPER.readValue(geoipServiceConfigYaml, GeoIpServiceConfig.class);
+
+        System.setProperty("data-prepper.dir", "build");
```

Comment:
This seems to overwrite the same system property set in build.gradle. But both places point to the same directory, so there is probably no issue.

Reply:
I originally had this line and then decided to set it in the `build.gradle`. So this line should be removed. Thanks!

Reply:
Actually, leaving this here will also help if you run the tests from IntelliJ directly. So I will leave it as-is.